Lib/test/test_json/test_tool.py

```diff
@@ -246,65 +248,72 @@ def test_broken_pipe_error(self):
-            ('false', b'\x1b[1;36mfalse\x1b[0m'),
-            ('NaN', b'NaN'),
-            ('Infinity', b'Infinity'),
-            ('-Infinity', b'-Infinity'),
-            ('"foo"', b'\x1b[1;32m"foo"\x1b[0m'),
-            (r'" \"foo\" "', b'\x1b[1;32m" \\"foo\\" "\x1b[0m'),
-            ('"α"', b'\x1b[1;32m"\\u03b1"\x1b[0m'),
-            ('123', b'123'),
-            ('-1.2345e+23', b'-1.2345e+23'),
+            ('{}', '{}'),
+            ('[]', '[]'),
+            ('null', f'{t.keyword}null{t.reset}'),
+            ('true', f'{t.keyword}true{t.reset}'),
+            ('false', f'{t.keyword}false{t.reset}'),
+            ('NaN', f'{t.number}NaN{t.reset}'),
+            ('Infinity', f'{t.number}Infinity{t.reset}'),
+            ('-Infinity', f'{t.number}-Infinity{t.reset}'),
+            ('"foo"', f'{t.string}"foo"{t.reset}'),
+            (r'" \"foo\" "', f'{t.string}" \\"foo\\" "{t.reset}'),
+            ('"α"', f'{t.string}"\\u03b1"{t.reset}'),
+            ('123', f'{t.number}123{t.reset}'),
+            ('-1.2345e+23', f'{t.number}-1.2345e+23{t.reset}'),
             (r'{"\\": ""}',
-             b'''\
-{
-    \x1b[94m"\\\\"\x1b[0m: \x1b[1;32m""\x1b[0m
-}'''),
+             f'''\
+{ob}
+    {t.definition}"\\\\"{t.reset}: {t.string}""{t.reset}
+{cb}'''),
             (r'{"\\\\": ""}',
-             b'''\
-{
-    \x1b[94m"\\\\\\\\"\x1b[0m: \x1b[1;32m""\x1b[0m
-}'''),
+             f'''\
+{ob}
+    {t.definition}"\\\\\\\\"{t.reset}: {t.string}""{t.reset}
+{cb}'''),
             ('''\
 {
     "foo": "bar",
     "baz": 1234,
     "qux": [true, false, null],
     "xyz": [NaN, -Infinity, Infinity]
 }''',
-             b'''\
-{
-    \x1b[94m"foo"\x1b[0m: \x1b[1;32m"bar"\x1b[0m,
-    \x1b[94m"baz"\x1b[0m: 1234,
-    \x1b[94m"qux"\x1b[0m: [
-        \x1b[1;36mtrue\x1b[0m,
-        \x1b[1;36mfalse\x1b[0m,
-        \x1b[1;36mnull\x1b[0m
+             f'''\
+{ob}
+    {t.definition}"foo"{t.reset}: {t.string}"bar"{t.reset},
+    {t.definition}"baz"{t.reset}: {t.number}1234{t.reset},
+    {t.definition}"qux"{t.reset}: [
+        {t.keyword}true{t.reset},
+        {t.keyword}false{t.reset},
+        {t.keyword}null{t.reset}
     ],
-    \x1b[94m"xyz"\x1b[0m: [
-        NaN,
-        -Infinity,
-        Infinity
+    {t.definition}"xyz"{t.reset}: [
+        {t.number}NaN{t.reset},
+        {t.number}-Infinity{t.reset},
+        {t.number}Infinity{t.reset}
     ]
-}'''),
+{cb}'''),
         )
 
         for input_, expected in cases:
             with self.subTest(input=input_):
                 with open(infile, "w", encoding="utf-8") as fp:
                     fp.write(input_)
-                _, stdout, _ = assert_python_ok('-m', self.module, infile,
-                                                PYTHON_COLORS='1')
-                stdout = stdout.replace(b'\r\n', b'\n')  # normalize line endings
+                _, stdout_b, _ = assert_python_ok(
+                    '-m', self.module, infile, FORCE_COLOR='1', __isolated='1'
```

Comment:
Maybe `True` is more explicit for the `__isolated` argument? (I didn't find any other tests that use `'1'`, it is common to use `True` or `False`)
```suggestion
                    '-m', self.module, infile, FORCE_COLOR='1', __isolated=True
```

Reply:
I used `'1'` on the basis that it's stored in and simulates an environment variable, and there's only strings in environment variables.